doc/reference/kernel/other/cpu_idle.rst

```diff
@@ -25,8 +25,8 @@ Making the CPU idle
 ===================
 
 Making the CPU idle is simple: call the k_cpu_idle() API. The CPU will stop
-executing instructions until an event occurs. Make sure interrupts are not
```

Comment:
This did not seem to be accurate any more @andrewboie @andyross to confirm :)

Reply:
This is... not our clearest interface.  The way k_cpu_idle() works is that it's:

1. Called with interrupts **disabled**
2. Expected not to return until an interrupt has occurred and been handled (though this isn't a firm requirement)
3. Returns with interrupts **enabled**.  No, I don't know why either.  No, it doesn't honor a irq_lock() key passed in from the caller, it just unmasks them unconditionally.

So removing that sentence is correct, but we should probably rewrite this section more fully.

Reply:
So, then, let's remove the sentence and complete the description as you propose :)

Reply:
Btw - this is how it works in the idle thread.

Reply:
@andyross i've done the following
- Added your point (3) in the description of k_cpu_idle example in this section. IMO, this section (paragraph) is only there to give an example of how k_cpu_idle() may work as a standalone API, i.e. _outside_ the context of the Zephyr idle thread. Your point (1) is specific to how the function is used in the idle thread.
- Extended the documentation of k_cpu_idle and arch_cpu_idle APIs with points (3) and (2), respectively. 
Let me know your thoughts about it.

@andrewboie ^^ as well; appreciate some feedback, as this touches an all-ARCH API

Reply:
This looks great to me.  I think the only argument against documenting this is that it makes it incrementally harder to fix.  I mean what we'd really **want** is for us to coordinate changing all the architectures to have simpler semantics (e.g. have it not muck with the interrupt mask state and just act like any other function that expects to be called with interrupts masked).

Reply:
> This looks great to me. I think the only argument against documenting this is that it makes it incrementally harder to fix. I mean what we'd really **want** is for us to coordinate changing all the architectures to have simpler semantics (e.g. have it not muck with the interrupt mask state and just act like any other function that expects to be called with interrupts masked).

@andyross then I will remove the added documentation in the function specification, to make it easier for you to change this later.

I now only added a note that "in some architectures" the function is unconditionally enabling IRQs upon return; that's needed to raise awareness of what's really happening and does not stop Kernel maintainers from changing the semantics, IMO.